model/clusters_mgmt/v1/cloud_available_regions_resource.model

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages the collection of available regions of a cloud provider.
+resource CloudAvailableRegions {
+	// Retrieves the list of regions of the cloud provider.
+	//
+	// IMPORTANT: This collection doesn't currently support paging or searching, so the returned
+	// `page` will always be 1 and `size` and `total` will always be the total number of regions
+	// of the provider.
+	method List {
+		// Index of the returned page, where one corresponds to the first page. As this
+		// collection doesn't support paging the result will always be `1`.
+		in out Page Integer = 1
+
+		// Number of items that will be contained in the returned page. As this collection
+		// doesn't support paging or searching the result will always be the total number of
+		// regions of the provider.
+		in out Size Integer = 100
+
+		// Total number of items of the collection that match the search criteria,
+		// regardless of the size of the page. As this collection doesn't support paging or
+		// searching the result will always be the total number of regions of the provider.
+		out Total Integer
+
+		// Retrieved list of cloud providers.
+		out Items []CloudAvailableRegion
```

Comment:
Maybe we could change this to `CloudRegion` and avoid having the duplicate type

Reply:
I was afraid that with CloudRegion, it will add the ../available_regions/{region_id} endpoint. 'make' still does not work for me in the sdk, so I cannot see the output of this. I will change this to CloudRegion so we can see if that works for us.

Reply:
updated the code. Endpoint should be /api/clusters_mgmt/v1/cloud_providers/{cloud_provider_id}/available_regions

Reply:
Very cool, this should technically make `AvailableRegion` obsolete and the _type file can be removed